Userland/Libraries/LibGUI/Desktop.cpp

```diff
@@ -59,13 +59,27 @@ RefPtr<Gfx::Bitmap> Desktop::wallpaper_bitmap() const
     return ConnectionToWindowServer::the().get_wallpaper().bitmap();
 }
 
-bool Desktop::set_wallpaper(RefPtr<Gfx::Bitmap const> wallpaper_bitmap, Optional<StringView> path)
+bool Desktop::apply_wallpaper(RefPtr<Gfx::Bitmap const> wallpaper_bitmap, Optional<StringView> path)
 {
     if (m_is_setting_desktop_wallpaper)
         return false;
 
+    if (!wallpaper_bitmap && path.has_value() && !path->is_empty()) {
+        auto maybe_wallpaper_bitmap = Gfx::Bitmap::load_from_file(*path, 1, rect().size());
```

Comment:
consider adding `auto const scale_factor = 1;` on the line above and passing that in to make the `1` look less mysterious

Reply:
Sure, done :+1: